state/pool.go

```diff
@@ -8,18 +8,49 @@ import (
 	"fmt"
 	"runtime/debug"
 	"sync"
+	"time"
 
 	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/pubsub"
 	"gopkg.in/juju/names.v2"
+	"gopkg.in/juju/worker.v1"
+
+	"github.com/juju/juju/state/watcher"
 )
 
+var errPoolClosed = errors.New("pool closed")
+
 // NewStatePool returns a new StatePool instance. It takes a State
 // connected to the system (controller model).
 func NewStatePool(systemState *State) *StatePool {
-	return &StatePool{
+	pool := &StatePool{
 		systemState: systemState,
 		pool:        make(map[string]*PoolItem),
+		hub:         pubsub.NewSimpleHub(nil),
+	}
+	// If systemState is nil, this is clearly a test, and a poorly
```

Comment:
Can we log a warning so that if we accidentally introduce nil system state in production code, we at least get to see an artefact

Reply:
sure